tests/io/test_data_catalog.py

```diff
@@ -266,6 +266,20 @@ def test_multi_catalog_list(self, multi_catalog):
         assert "abc" in entries
         assert "xyz" in entries
 
+    def test_multi_catalog_list_regex(self, multi_catalog):
+        """Test that regex patterns filter data sets accordingly"""
+        assert multi_catalog.list(regex_search="^a") == ["abc"]
+        assert multi_catalog.list(regex_search="a|x") == ["abc", "xyz"]
+        assert multi_catalog.list(regex_search="^(?!(a|x))") == []
+        assert multi_catalog.list(regex_search="def") == []
```

Comment:
You could parametrize this unit test to take regex filter and expected output.
Would it be worth including a case for empty string?

Reply:
Updated, good catch on empty string - I'm not sure I'd want it to match everything